assembly-ide/src/assembly/bin/che.sh

```diff
@@ -10,20 +10,436 @@
+
+  # Test to verify that docker can reach the VM
+  "${DOCKER}" ps &> /dev/null
+  DOCKER_VM_REACHABLE=$?
+
+  if [ $DOCKER_EXISTS -eq 1 ] || [ $DOCKER_VM_REACHABLE -eq 1 ]; then
+      error_exit "!!! Running 'docker' and 'docker ps' failed. Either docker is unreachable, or docker cannot reach your vbox VM. !!!"
+    return
+  fi
+
+  # EOF
+  if $win || $mac ; then
+    echo -e "${BLUE}Docker${NC} is configured to use vbox docker-machine named ${GREEN}$VM${NC} with IP ${GREEN}$("${DOCKER_MACHINE}" ip $VM)${NC}..."
+    echo
+  else
+    echo "Docker is natively installed and reachable..."
+  fi  
+}
+
+function strip_url {
+  # extract the protocol
+  proto="`echo $1 | grep '://' | sed -e's,^\(.*://\).*,\1,g'`"
+  
+  # remove the protocol
+  url=`echo $1 | sed -e s,$proto,,g`
+
+  # extract the user and password (if any)
+  userpass="`echo $url | grep @ | cut -d@ -f1`"
+  pass=`echo $userpass | grep : | cut -d: -f2`
+  
+  if [ -n "$pass" ]; then
+      user=`echo $userpass | grep : | cut -d: -f1`
+  else
+      user=$userpass
+  fi
+
+  # extract the host and remove the port
+  hostport=`echo $url | sed -e s,$userpass@,,g | cut -d/ -f1`
+  port=`echo $hostport | grep : | cut -d: -f2`
+  if [ -n "$port" ]; then
+      host=`echo $hostport | grep : | cut -d: -f1`
+  else
+      host=$hostport
+  fi
+
+  # extract the path (if any)
+  path="`echo $url | grep / | cut -d/ -f2-`"
+
+}
+
+function print_client_connect {
+  if ! $USE_DOCKER ; then 
+    echo -e "
+############## HOW TO CONNECT YOUR CHE CLIENT ###############
+After Che server has booted, you can connect your clients by:
+1. Open browser to http://localhost:${USE_PORT}, or:
+2. Open native chromium app.
+#############################################################\n\n"
+  else 
+    echo -e "
+############## HOW TO CONNECT YOUR CHE CLIENT ###############
+After Che server has booted, you can connect your clients by:
+1. Open browser to http://${host}:${USE_PORT}, or:
+2. Open native chromium app.
+#############################################################\n\n"
+
+  fi
+}
+
+function call_catalina {
+
+  # Test to see that Che application server is where we expect it to be
+  if [ ! -d "${ASSEMBLY_BIN_DIR}" ]; then
+    error_exit
+    return
+  fi
+
+  ### Cannot add this in setenv.sh.
+  ### We do the port mapping here, and this gets inserted into server.xml when tomcat boots
+  [ -z "${JAVA_OPTS}" ]  && export JAVA_OPTS="-Dport.http=${USE_PORT}"
+  #Class path
+  [ -z "${SERVER_PORT}" ]  && export SERVER_PORT=${USE_PORT}
+
+  # Launch the Che application server, passing in command line parameters
+  ${ASSEMBLY_BIN_DIR}/catalina.sh ${USE_SERVER_ACTION}
+
+}
+
+function stop_che_server {
+
+  if ! $USE_DOCKER; then
+    echo -e "Stopping Che server running on localhost:${USE_PORT}"
+    call_catalina >/dev/null 2>&1
+  else
+    echo -e "Stopping Che server running in docker container."
+
+    DOCKER_EXEC="sudo service docker stop && /home/user/che/bin/che.sh stop"
```

Comment:
It is the location of Tomcat inside the Docker image running the Che server.

-Tyler




On Tue, Dec 29, 2015 at 12:48 PM -0800, "Sergii Kabashniuk" <notifications@github.com> wrote:












In assembly-ide/src/assembly/bin/che.sh:
> +  [ -z "${SERVER_PORT}" ]  && export SERVER_PORT=${USE_PORT}
> +
> +  # Launch the Che application server, passing in command line parameters
> +  ${ASSEMBLY_BIN_DIR}/catalina.sh ${USE_SERVER_ACTION}
> +
> +}
> +
> +function stop_che_server {
> +
> +  if ! $USE_DOCKER; then
> +    echo -e "Stopping Che server running on localhost:${USE_PORT}"
> +    call_catalina >/dev/null 2>&1
> +  else
> +    echo -e "Stopping Che server running in docker container."
> +
> +    DOCKER_EXEC="sudo service docker stop && /home/user/che/bin/che.sh stop"



what is /home/user? where it's located?



—
Reply to this email directly or view it on GitHub.